File: app/src/main/java/com/github/npcdw/store/util/HttpUtil.java
package com.github.npcdw.store.util;

import com.github.npcdw.store.config.GlobalConfig;

import java.io.IOException;

import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

/**
 * Http接口工具类
 */
public class HttpUtil {
    public static final MediaType JSON = MediaType.get("application/json; charset=utf-8");

    public static String post(String url, String body) throws IOException {
        RequestBody requestBody = RequestBody.create(body, JSON);
        Request request = new Request.Builder()
                .url(url)
                .post(requestBody)
                .header("token", GlobalConfig.TOKEN)
                .build();
        OkHttpClient client = new OkHttpClient();
        try (Response response = client.newCall(request).execute()) {
            if (response.code() != 200 || response.body() == null) {
                return null;
            }
            return response.body().string();
        }
    }

    public static String get(String url) throws IOException {
        Request request = new Request.Builder()
                .url(url)
                .header("token", GlobalConfig.TOKEN)
                .build();
        OkHttpClient client = new OkHttpClient();
        try (Response response = client.newCall(request).execute()) {
            if (response.code() != 200 || response.body() == null) {
                return null;
            }
            return response.body().string();
        }
    }

    public static String put(String url, String body) throws IOException {
        RequestBody requestBody = RequestBody.create(body, JSON);
        Request request = new Request.Builder()
                .url(url)
                .put(requestBody)
                .header("token", GlobalConfig.TOKEN)
                .build();
        OkHttpClient client = new OkHttpClient();
        try (Response response = client.newCall(request).execute()) {
            if (response.code() != 200 || response.body() == null) {
                return null;
            }
            return response.body().string();
        }
    }

    public static String delete(String url) throws IOException {
        Request request = new Request.Builder()
                .url(url)
                .delete()
                .header("token", GlobalConfig.TOKEN)
                .build();
        OkHttpClient client = new OkHttpClient();
        try (Response response = client.newCall(request).execute()) {
            if (response.code() != 200 || response.body() == null) {
                return null;
            }
            return response.body().string();
        }
    }

}

File: app/src/main/java/com/github/npcdw/store/util/HttpUtil.kt
package com.github.npcdw.store.util

import com.github.npcdw.store.config.GlobalConfig
import okhttp3.MediaType
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.RequestBody.Companion.toRequestBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody
import java.io.IOException

/**
 * Http接口工具类
 */
object HttpUtil {
    private val APPLICATION_JSON: MediaType = "application/json; charset=utf-8".toMediaType()

    @Throws(IOException::class)
    fun post(url: String, body: String): String? {
        val requestBody: RequestBody = body.toRequestBody(APPLICATION_JSON);
        val request: Request = Request.Builder()
            .url(url)
            .post(requestBody)
            .header("token", GlobalConfig.TOKEN)
            .build()
        val client = OkHttpClient()
        client.newCall(request).execute().use { response ->
            return if (response.code != 200 || response.body == null) {
                null
            } else response.body!!.string()
        }
    }

    @Throws(IOException::class)
    operator fun get(url: String): String? {
        val request: Request = Request.Builder()
            .url(url)
            .header("token", GlobalConfig.TOKEN)
            .build()
        val client = OkHttpClient()
        client.newCall(request).execute().use { response ->
            return if (response.code != 200 || response.body == null) {
                null
            } else response.body!!.string()
        }
    }

    @Throws(IOException::class)
    fun put(url: String, body: String): String? {
        val requestBody: RequestBody = body.toRequestBody(APPLICATION_JSON);
        val request: Request = Request.Builder()
            .url(url)
            .put(requestBody)
            .header("token", GlobalConfig.TOKEN)
            .build()
        val client = OkHttpClient()
        client.newCall(request).execute().use { response ->
            return if (response.code != 200 || response.body == null) {
                null
            } else response.body!!.string()
        }
    }

    @Throws(IOException::class)
    fun delete(url: String): String? {
        val request: Request = Request.Builder()
            .url(url)
            .delete()
            .header("token", GlobalConfig.TOKEN)
            .build()
        val client = OkHttpClient()
        client.newCall(request).execute().use { response ->
            return if (response.code != 200 || response.body == null) {
                null
            } else response.body!!
                .string()
        }
    }
}
